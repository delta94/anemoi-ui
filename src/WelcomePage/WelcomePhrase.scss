@import "../variables";

.WelcomePhrase {
	color: white;
}

.WelcomePhraseIntroduction {
	font-weight: bold;
}

.WelcomePhrase em {
	color: $indigo;
	font-style: normal;
}
