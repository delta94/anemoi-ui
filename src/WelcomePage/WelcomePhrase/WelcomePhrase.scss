@import "../../variables";

.WelcomePhrase {
	color: white;

	& em {
		transition: color 1.5s ease-in;
		color: $indigo;
		font-style: normal;
	}

	&.Dark em {
		transition: color 1.5s ease-in;
		color: $blue;
	}
}

.WelcomePhraseIntroduction {
	font-weight: bold;
}
