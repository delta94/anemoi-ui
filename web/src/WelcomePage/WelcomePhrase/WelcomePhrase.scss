@import "../../variables";

.WelcomePhrase {
	color: white;

	& em {
		transition: color 1s ease-in;
		color: $indigo;
		font-style: normal;
	}

	&.Dark em {
		transition: color 1s ease-in;
		color: $blue;
	}
}

.WelcomePhraseIntroduction {
	font-weight: bold;
}
