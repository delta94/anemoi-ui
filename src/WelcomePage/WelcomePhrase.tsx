import React from 'react';

import './WelcomePhrase.scss';
import { Col, Row, Button } from 'react-bootstrap';

interface WelcomePhraseProps {
    collapse: Function
}
const WelcomePhrase: React.FC<WelcomePhraseProps> = (props: WelcomePhraseProps) => {
    return (
        <Col xs={{ span: 12, order: 2 }} md={{ span: 6, order: 1 }}>
            <Row className="h-100">
                <Col xs={{ span: 10, offset: 1 }} className="my-auto">
                    <div className="WelcomePhrase">
                        <h1 className="WelcomePhraseIntroduction">
                            Bem-vindo, eu sou o <em>Anemoi.</em>
                            <br />
                            Expert em <em>viagens.</em>
                            <br />
                        </h1>
                        <h1 className="WelcomePhraseCallToAction">
                            Possor ajudar a planejar a sua?
                    </h1>
                    </div>
                    <Button size="lg" variant="primary" onClick={() => props.collapse()}>
                        <b>Sim!</b>
                    </Button>
                </Col>
            </Row>
        </Col>
    );
}

export default WelcomePhrase;
