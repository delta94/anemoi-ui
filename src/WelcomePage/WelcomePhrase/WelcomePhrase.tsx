import React, { useContext } from 'react';

import './WelcomePhrase.scss';
import { useTranslation, Trans } from 'react-i18next';
import { ThemeContext } from '../../Shared/ThemeContext';

const WelcomePhrase: React.FC = () => {
  const { t } = useTranslation();
  const themeContext = useContext(ThemeContext);

  return (
    <div className={`WelcomePhrase ${themeContext.theme === "DARK" ? "Dark" : ""}`}>
      <h1 className="WelcomePhraseIntroduction">
        <Trans i18nKey="WELCOME_IM_ANEMOI">
          Welcome I am <em>Anemoi.</em>
        </Trans>
        <br />
        <Trans i18nKey="TRAVEL_EXPERT">
          Travel <em>expert.</em>
        </Trans>
        <br />
      </h1>
      <h1 className="WelcomePhraseCallToAction">{t('CAN_I_HELP_PLANNING_YOURS')}</h1>
    </div>
  );
};

export default WelcomePhrase;
