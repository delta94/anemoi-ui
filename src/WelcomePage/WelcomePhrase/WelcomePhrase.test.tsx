import React from 'react';

import { render, cleanup } from '@testing-library/react';
import WelcomePhrase from './WelcomePhrase';

describe('WelcomePage component', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders without crashing', () => {
    const { asFragment } = render(<WelcomePhrase />);

    expect(asFragment()).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
    expect(asFragment().querySelector('.WelcomePhrase')).toBeTruthy();
  });
});
