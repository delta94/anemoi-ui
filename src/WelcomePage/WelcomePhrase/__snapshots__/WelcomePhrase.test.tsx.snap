// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePage component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="WelcomePhrase"
  >
    <h1
      class="WelcomePhraseIntroduction"
    >
      Welcome I am 
      <em>
        Anemoi.
      </em>
      <br />
      Travel 
      <em>
        expert.
      </em>
      <br />
    </h1>
    <h1
      class="WelcomePhraseCallToAction"
    >
      CAN_I_HELP_PLANNING_YOURS
    </h1>
  </div>
</DocumentFragment>
`;
